File: 05_hash_tables/swift/01_price_of_groceries.swift
import Foundation

var book = [String: Double]()
// an apple costs 67 cents
book["apple"] = 0.67
//# milk costs $1.49
book["milk"] = 1.49
book["avacado"] = 1.49
print(book) // => ["avacado": 1.49, "apple": 0.67000000000000004, "milk": 1.49]

// Qustion: Why is "apple" is 0.67000000000000004 intsted of 0.67?
// Answer: Double cannot store the value 0.67 exactly. Swift uses (like many other languages) binary floating point numbers according to the IEEE 754 standard.
// This topic is not related with Algorithms, but you can play with .description and .debugDescription for making workarrounds 
print(book["apple"]?.description ?? "Not exist") // => 0.67
print(book["apple"]?.debugDescription ?? "Not exist") // => 0.67000000000000004

File: 05_hash_tables/golang/01_price_of_groceries.go
package main

import "fmt"

var book map[string]float64

func main() {
	book = make(map[string]float64)
	book["apple"] = 0.67
	book["milk"] = 1.49
	book["avocado"] = 1.49
	fmt.Println(book)
}
